### PR TITLE
query decoy result

### PR DESCRIPTION
**Pull request by timosachsenberg**
# Description

Please include a summary of the change and which issue is fixed.

# Checklist:
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

Note: If your PR is failing you can check out http://cdash.openms.de/index.php?project=OpenMS and look for your PR with detailed error messages.

### CODE REVIEW

File: src/topp/MascotAdapterOnline.cpp [hunk 1]
```diff
@@ -257,28 +265,59 @@ class TOPPMascotAdapterOnline :
...
+        {
+          mascot_tmp_file.remove(); // delete file
+        }
       }
-      else
+
+      if (internal_decoys)
       {
-        mascot_tmp_file.remove(); // delete file
+        String mascot_tmp_file_name(File::getTempDirectory() + "/" + File::getUniqueName() + "_Mascot_decoy_response");
```
Comment (cbielow) [inline src/topp/MascotAdapterOnline.cpp:296]: this section is too much copy-paste.
Use a function?

File: src/topp/MascotAdapterOnline.cpp [hunk 2]
```diff
@@ -257,28 +265,59 @@ class TOPPMascotAdapterOnline :
...
+        mascot_tmp_file.write(mascot_query->getMascotXMLResponse());
+        mascot_tmp_file.close();
+      
+        // set up helper object for looking up spectrum meta data:
+        SpectrumMetaDataLookup lookup;
+        MascotXMLFile::initializeLookup(lookup, exp);
+
+        // read the response
+        vector<PeptideIdentification> pep_ids_decoy;
+        ProteinIdentification prot_id_decoy;
```
Comment (cbielow) [inline src/topp/MascotAdapterOnline.cpp:308]: ehh.... do you actually use the decoy result after loading it?

File: src/openms/source/FORMAT/MascotRemoteQuery.cpp [hunk 3]
```diff
@@ -579,13 +579,27 @@ namespace OpenMS
 
       // see http://www.matrixscience.com/help/export_help.html for parameter documentation
       String required_params = "&do_export=1&export_format=XML&generate_file=1&group_family=1&peptide_master=1&protein_master=1&search_master=1&show_unassigned=1&show_mods=1&show_header=1&show_params=1&prot_score=1&pep_exp_z=1&pep_score=1&pep_seq=1&pep_homol=1&pep_ident=1&pep_expect=1&pep_var_mod=1&pep_scan_title=1&query_qualifiers=1&query_peaks=1&query_raw=1&query_title=1";
+      // TODO: check that export_params don't contain _show_decoy_report=1. This would add <decoy> at the top of the XML document and we can't easily distinguish the two files (target/decoy) from the search results later.
```
Comment (cbielow) [inline src/openms/source/FORMAT/MascotRemoteQuery.cpp:582]: worth mentioning that in the docs of `export_params` flag and maybe actually do the check?
Its only a single line of code.

File: src/topp/MascotAdapterOnline.cpp (hunk 2 above)
Comment (timosachsenberg) [reply src/topp/MascotAdapterOnline.cpp:308]: No I am even not sure if it works if two different requests are processed async. I was hoping that someone who worked with MAO knows if that could work

**Comment by timosachsenberg** [comment]
I think I found a way. I now export decoys after target results are fully downloaded (doing it asynchronously while the other query was still running (potentially going through several redirects) did not work.

File: src/openms/source/FORMAT/MascotRemoteQuery.cpp (hunk 3 above)
Comment (timosachsenberg) [reply src/openms/source/FORMAT/MascotRemoteQuery.cpp]: I could add it thou it doesn't help a lot as a user should not mess with the export_params anyway. Basically if you change them in any way (not only related to the decoys) you will likely break the export, so you must know in detail what you are doing.

**Review by jpfeuffer — APPROVED** [review]

**Pull request merged**